### PR TITLE
Fix HTML errors generated by EmailableReporter2.

### PR DESCRIPTION
**Pull request by rankinc**
With these changes, the W3C HTML Validator tool (validator.w3.org) validates my emailable-report.html file as XHTML 1.1. Previously, the report generated incorrect HTML for empty test summaries and results. Also, the "back to summary" links in the results section now return the user to the top of the summary table again.

### CODE REVIEW

**Comment by krmahadevan** [comment]
@juherr - I tried validating the contents of `emailable-report.html` that is currently generated against https://jigsaw.w3.org/css-validator/

I didn't see any errors. I guess we can close off this PR (Just trying to see what we can clean up in terms of the open PRs)

Attaching a zip version of the `emailable-report.html` for reference.

[emailable-report.html.zip](https://github.com/cbeust/testng/files/818508/emailable-report.html.zip)

**Comment by juherr** [comment]
@krmahadevan I've just checked your file on https://validator.w3.org and I see:

> 18 Errors, 2 warning(s)

**Comment by krmahadevan** [comment]
@juherr then let me try against that site with this PR changes. I wasn't sure which site to use for validation.
